### PR TITLE
Check varargs null values in JWTVerifier

### LINKED ISSUE #404: Issue on setting audience/issuer as var-args (JWTVerifier)
**Please do not report security vulnerabilities here**. The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.

**Thank you in advance for helping us to improve this library!** Please read through the template below and answer all relevant questions. Your additional work here is greatly appreciated and will help us respond as quickly as possible. For general support or usage questions, use the [Auth0 Community](https://community.auth0.com/) or [Auth0 Support](https://support.auth0.com/). Finally, to avoid duplicates, please search existing Issues before submitting one here.

By submitting an Issue to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

> when sending null to withAudience/withIssuer (JWTVerifier), varargs creates an array instead of null. Due to that even though there is no "aud/iss" value, it creates claim and it tries to match.

### Reproduction

https://stackoverflow.com/questions/4028059/calling-java-varargs-method-with-single-null-argument/4028113

### Environment

> Please provide the following:

- **Version of this library used:** 3.8.3
- **Version of the platform or framework used, if applicable:**
- **Other relevant versions (language, server software, OS, browser):**
- **Other modules/plugins/libraries that might be involved:**

### PR DESCRIPTION
**Pull request by jimmyjames**
# Changes

Fixes #404 

- Check for null values on issuer and audience varargs arguments.
- Also fixes test compile warnings for ambiguous null to vararg APIs.

### References

- #404
- #405 

### Testing

- [X] This change adds test coverage

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: lib/src/main/java/com/auth0/jwt/JWTVerifier.java [hunk 1]
```diff
@@ -230,6 +230,20 @@ private void requireClaim(String name, Object value) {
         }
     }
 
+    private static boolean checkVarargNull(String[] args) {
```
Comment (damieng) [inline lib/src/main/java/com/auth0/jwt/JWTVerifier.java]: Should this be `checkVarArgNull` ?

Maybe a better name might be isNullOrEmpty?

Comment (jimmyjames) [reply lib/src/main/java/com/auth0/jwt/JWTVerifier.java]: Renamed to `isNullOrEmpty`
